File: l33tcode/online-stock-span.py
class StockSpanner:
    def __init__(self):
        self._stack = []

    def next(self, price: int) -> int:
        weight = 1

        while self._stack and self._stack[-1][0] <= price:
            weight += self._stack.pop()[1]

        self._stack.append((price, weight))

        return weight


# Your StockSpanner object will be instantiated and called as such:
# obj = StockSpanner()
# param_1 = obj.next(price)

File: l33tcode/online-stock-span.go
package main

type StackItem struct {
	price  int
	weight int
}

type StockSpanner struct {
	stack []StackItem
}

func Constructor() StockSpanner {
	return StockSpanner{stack: []StackItem{}}
}

func (this *StockSpanner) Next(price int) int {
	weight := 1
	for len(this.stack) > 0 && this.stack[len(this.stack)-1].price <= price {
		weight += this.stack[len(this.stack)-1].weight
		this.stack = this.stack[:len(this.stack)-1]
	}
	this.stack = append(this.stack, StackItem{price: price, weight: weight})

	return weight
}

/**
 * Your StockSpanner object will be instantiated and called as such:
 * obj := Constructor();
 * param_1 := obj.Next(price);
 */
